packages/runner/injection/patches/cookies.ts

```diff
@@ -88,26 +95,46 @@ export const patchDocumentCookie = (requestCookies: AutomationCookie[]) => {
       const stringValue = `${newValue}`
       const parsedCookie = CookieJar.parse(stringValue)
 
+      if (parsedCookie?.hostOnly === null) {
+        // we want to make sure the hostOnly property is respected when syncing with CDP/extension to prevent duplicates.
+        // in the case it is not set, we need to calculate it
+        parsedCookie.hostOnly = isHostOnlyCookie(parsedCookie.domain || domain)
+      }
+
       // if result is undefined, it was invalid and couldn't be parsed
       if (!parsedCookie) return getDocumentCookieValue()
 
-      // we should be able to pass in parsedCookie here instead of the string
-      // value, but tough-cookie doesn't recognize it using an instanceof
-      // check and throws an error. because we can't, we have to massage
-      // some of the properties below to be correct
-      const cookie = setCookie(stringValue)
-
-      if (cookie) {
-        cookie.sameSite = parsedCookie.sameSite
-
-        if (!parsedCookie.path) {
-          cookie.path = '/'
+      // if the cookie is not expired, don't set it
+      if (parsedCookie.expiryTime() < Date.now()) {
+        cookieJar.removeCookie({
+          name: parsedCookie.key,
+          path: parsedCookie.path || '/',
+          domain: parsedCookie.domain as string,
+        })
+
+        // send the cookie to the server so it can be removed from the browser
+        // via aututomation. If the cookie expiry is set inside the server-side cookie jar,
+        // the cookie will be automatically removed.
+        sendCookieToServer(toughCookieToAutomationCookie(parsedCookie, domain))
+      } else {
```

Comment:
Shoudl be able to return and skip adding everything to the else block
```js
if (expired) {
  // remove
  return
}
// add cookie logic
```
I tend to find it easier to read

Reply:
fixed in a67c06d4b4